examples-kotlin/src/main/kotlin/org/bitcoinj/ForwardingService.kt

```diff
@@ -114,72 +120,61 @@ class ForwardingService(val config: Config, val wallet: Wallet) : Closeable {
+                }
+                else {
+                    println("Exception occurred: $throwable")
+                }
+            }
     }
 
-    /**
-     * Create a CoinSelector that only returns outputs from a given parent transaction.
-     *
-     *
-     * This is using the idea of partial function application to create a 2-argument function for coin selection
-     * with a third, fixed argument of the transaction id.
-     * @param parentTxId The parent transaction hash
-     * @return a coin selector
-     */
-    fun forwardingCoinSelector(parentTxId: Sha256Hash): CoinSelector {
-        requireNonNull(parentTxId)
-        return CoinSelector.fromPredicate { output: TransactionOutput ->
-            output.parentTransactionHash == parentTxId
-        }
+    fun status(): String? {
+        return String.format("Waiting to receive coins on: %s", wallet.currentReceiveAddress())
     }
 
     /**
      * Close the service.
      */
     override fun close() {
         wallet.removeCoinsReceivedEventListener(::coinForwardingListener)
+        walletAppKit.close()
     }
 }
 
-fun getPrefix(network: BitcoinNetwork?): String {
-    return String.format("forwarding-service-%s", network.toString())
-}
-
-fun parseArgs(args: Array<String>): Array<Any> {
-    val USAGE = "Usage: address-to-forward-to [mainnet|testnet|signet|regtest]"
-    if (args.size < 1 || args.size > 2) {
-        System.err.println(USAGE)
-        System.exit(1)
-    }
-    val network: BitcoinNetwork
-    val forwardingAddress: Address
-    if (args.size >= 2) {
-        // If network was specified, validate address against network
-        network = BitcoinNetwork.fromString(args[1]).orElseThrow()
-        forwardingAddress = AddressParser.getDefault(network).parseAddress(args[0])
-    } else {
-        // Else network not-specified, extract network from address
-        forwardingAddress = AddressParser.getDefault().parseAddress(args[0])
-        network = forwardingAddress.network() as BitcoinNetwork
+data class Config(
+    val forwardingAddress: Address,                                              // Address to forward to
+    val network: BitcoinNetwork = forwardingAddress.network() as BitcoinNetwork, // Network to operate on
+    val walletDirectory: File,                                                   // Directory to create wallet files in
+    val walletPrefix: String = getPrefix(network),                               // Prefix for wallet file names
+    val requiredConfirmations: Int = 1,                                          // Required number of tx confirmations before forwarding
+    val maxConnections: Int = 4                                                  // Maximum number of connections
+) {
+    constructor(forwardingAddress: Address) : this(
+        forwardingAddress,
+        forwardingAddress.network() as BitcoinNetwork
+    )
```

Comment:
Is this constructor necessary?

Reply:
Yes, I noticed  you do the address check in a second constructor that has (address, network) as parameters. So this constructor for when only address is passed, calls the other constructor that checks address before address is set on member variables. 

since forwardingAddress.network() as BitcoinNetwork is now being done in the constructor, i should also take it out of the default set value for network